openapi/README.md

```diff
@@ -0,0 +1,40 @@
+# Decomposed OpenAPI Structure
+
+The goal of decomposing the OpenAPI YAML file is to simplify adding new resources. Each
+resource can be added nearly independently.
+
+In the initial implementation, all files in all of the subdirectories of `src/` are scanned by the tool
+and merged into a single YAML output file. That output is processed by the swagger codegen
+tool to generate the client and server code.
+
+## Directory Structure
+
+The directory structure is setup like this:
+- parts
+- resources
+- common
+- - schemas.yaml - shared schema elements
+- - responses.yaml - shared response elements
+- - parameters.yaml - shared parameter elements
+
+### Parts
+This directory contains all of the paths and the high-level schema material.
+
+### Resources
+This directory contains the resource and attribute definitions for all resource types.
+Many resources share their attribute and resource definitions between a controlled and a
+referenced version. Those definitions are also referenced by the enumeration
+endpoints. Therefore, we place all resource attribute and definition files here.
+
+In the future, we may drive auto-generation of enumeration unions by scanning this
+directory. Files should be named:
+```
+resource_{resource-type}
+```
+For example, the file describing a GCP GCS bucket would be `resource_gcp_gcs_bucket.yaml`.
```

Comment:
Why not have a `resource ` subdirectory so you don't have confusion of which underscore is a delimieter. If you don't want to do that, consider `resource__gcp_gcs_bucket.yaml`.

Reply:
It is redundant, since they are in a `resources` directory. I'll rename them.